odl/trafos/fourier.py

```diff
@@ -1365,13 +1366,25 @@ def _call_pyfftw(self, x, out, **kwargs):
         return out
 
     @property
+    def adjoint(self):
+        """The adjoint Fourier transform."""
+        if self.domain.exponent != 2.0 or self.range.exponent != 2.0:
+            raise NotImplementedError(
+                'Adjoint only implemented for exponent=2')
```

Comment:
At some point I was considering scrapping these checks. Why should we perform them? In a generalized sense the adjoint relation is still correct even if `p != 2`, so why forbidding it? Just a thought (again).